### PR TITLE
Add associate() to sequelize models

### PR DESCRIPTION
**Pull request by Qard**
This adds a helpful `associate()` function to sequelize models which makes describing relational connections more straight forward.

### CODE REVIEW

File: generators/app/templates/service.js [hunk 1]
```diff
@@ -23,4 +23,12 @@ module.exports = function() {
   <% if (localAuth || authentication.length) { %>
   app.configure(authentication);<% } %><% for (var i = 0; i < services.length; i++) { %>
   app.configure(<%= services[i] %>);<% } %>
+  <% if (database === 'sqlite' || database === 'mssql' || database === 'postgres' || database === 'mysql' || database === 'mariadb') { %>
+  const {models} = sequelize;
+  app.set('models', models);
+  Object.keys(models).forEach(name => {
+    if ('associate' in models[name]) {
```
Comment (daffl) [inline generators/app/templates/service.js]: Where is `associate` coming from?

Comment (Qard) [inline generators/app/templates/service.js]: from `classMethods.associate` in `generators/model/templates/sequelize.js`. This change would include an empty `associate` function on sequelize models which get called after all the models are registered, that way the model load order doesn't matter.

I noticed after making this PR that this same design was already in discussion in #94. They *did* bring up the point though that the model.sync() bits would also need to be moved until *after* these `associate` functions run though.

**Comment by ekryski** [comment]
Given that the generators are getting an overhaul right now and are likely to be done soon. Let's keep this open for reference but this won't get merged into this branch. Thanks for submitting the PR @Qard! 😄

**Comment by daffl** [comment]
This has been added in v1.0.0-pre.1 via #159

**Pull request closed without merging**